src/parseline.c

```diff
@@ -838,27 +839,15 @@ void m_nick (char *sender, char *tail)
     uptr2 = find_user(newnick);
     if (uptr2) {
         if (uptr && (find_link2(oldnick, newnick) || find_link2(newnick, oldnick)) && uptr->authed == 1) {
-            uptr->authed = 0;
-            uptr->lastseen = time(NULL);
-            uptr2->authed = 1;
             SendRaw("SVSMODE %s +r",newnick);
             uptr2->lastseen = time(NULL);
         } else {
-            if (uptr && uptr->authed == 1)
-                SendRaw("SVSMODE %s -r",newnick);
             NoticeToUser(nptr, "This nick is registered. Please identify yourself or take another nick.");
```

Comment:
Can the user actually do that? Will re-identification work if already identified?